addons/website/static/src/snippets/s_popup/000.js

```diff
@@ -397,7 +405,98 @@ publicWidget.registry.cookies_bar = PopupWidget.extend({
             }
         }
         setUtmsHtmlDataset();
-    }
+    },
+    /**
+     * Reopens the cookies bar if it was closed.
+     *
+     * @private
+     */
+    _onShowCookiesBar() {
+        const modalEl = this.el.querySelector(".modal");
+        const currCookie = cookie.get(this.el.id);
+
+        if (currCookie && JSON.parse(currCookie).optional || !this._popupAlreadyShown) {
+            return;
+        }
+        $(modalEl).modal("show");
+
+        // The cookies bar remains hidden, most probably because of the browser
+        // or an extension: notify the user because "nothing happens when I
+        // click" is never good.
+        if (!isVisible(modalEl)) {
+            window.alert(_t("Our cookies bar was blocked by your browser or an extension."));
+            return;
+        }
+        modalEl.focus();
+    },
+});
+
+publicWidget.registry.cookiesApproval = publicWidget.Widget.extend({
+    selector: "[data-need-cookies-approval]",
+    events: {
+        "add_cookies_warning": "_onAddCookiesWarning",
+    },
+
+    /**
+     * @override
+     */
+    willStart() {
+        const thirdPartyEl = this.el.tagName === "IFRAME"
+            ? this.el
+            : this.el.querySelector("iframe");
+        if (thirdPartyEl) {
+            const existingOptionalCookiesWarningEl = thirdPartyEl.nextElementSibling
+                ?.classList.contains("o_no_optional_cookie");
+            if (!existingOptionalCookiesWarningEl) {
+                this._addOptionalCookiesWarning(thirdPartyEl);
+            }
+        }
+        return this._super(...arguments);
+    },
+    /**
+     * Adds a warning in place of the iframe. On click, shows the cookies bar if
+     * it was hidden.
+     *
+     * @private
+     */
+    _addOptionalCookiesWarning(iframeEl) {
+        const options = {
+            extraStyle: iframeEl.parentElement.classList.contains("media_iframe_video")
+                ? `aspect-ratio: 16/9; max-width: ${MEDIAS_BREAKPOINTS[SIZES.SM].maxWidth}px;`
+                : "",
+            extraClasses: getComputedStyle(iframeEl.parentElement).position === "absolute"
+                ? "" : "my-3",
+        };
+        const optionalCookiesWarningEl = renderToElement("website.cookiesWarning", options);
+        iframeEl.insertAdjacentElement("afterend", optionalCookiesWarningEl);
+        iframeEl.classList.add("d-none");
+        optionalCookiesWarningEl.addEventListener("click", () => {
+            $(document.getElementById("website_cookies_bar")).trigger("show_cookies_bar");
+        });
+
+        document.addEventListener("optionalCookiesAccepted", () => {
```

Comment:
Another event listener that should be removed in destroy, at least for consistency :pray: